gdk4/src/subclass/paintable.rs

```diff
@@ -34,7 +34,13 @@ pub trait PaintableImpl: ObjectImpl {
         self.parent_intrinsic_aspect_ratio(paintable)
     }
 
-    fn snapshot(&self, paintable: &Self::Type, snapshot: &Snapshot, width: f64, height: f64);
+    fn snapshot(
+        &self,
+        paintable: &Self::Type,
+        snapshot: &impl IsA<Snapshot>,
```

Comment:
Does this make any difference actually? It only makes things harder to use or not?

Reply:
See the paintable demo changes, you don't have to downcast and you can go through the AsRef trait

Reply:
But if you made this an `&gdk::Snapshot` then you wouldn't even have to call `as_ref()` and could do exactly the same things with your other changes.